.github/workflows/packer-build-and-publish.yml

```
@@ -150,75 +210,99 @@ jobs:
   publish:
     runs-on: ubuntu-latest
     needs:
-      - check-if-main
+      - check-main
+      - check-release
       - build
+    outputs:
+      release_id: ${{ steps.publish.outputs.id }}
+      release_created: ${{ steps.publish.outputs.id != '' }}
     steps:
       - name: Checkout repository
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
 
-      - name: Download artifacts
-        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+      # If a release already exists, copy assets from the existing release
+      - name: Get assets from existing release
+        if: ${{ needs.check-release.outputs.release_found == 'true' }}
```

nit:

```suggestion
        if: needs.check-release.outputs.release_found == 'true'
```

(there are some other examples)

---

Didn't realize those weren't needed for those conditions.